### PR TITLE
New attribute module for redmine project ID

### PR DESCRIPTION
**Pull request by balcirakpeter**

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_redmineProjectID.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * Module for redmine project ID
+ * 
+ * @author Peter Balcirak <peter.balcirak@gmail.com>
+ * @date 8.3.2016
+ */
+public class urn_perun_resource_attribute_def_def_redmineProjectID extends ResourceAttributesModuleAbstract implements ResourceAttributesModuleImplApi {
+
+        @Override
```
Comment (licehammer) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_redmineProjectID.java]: Please use tab for indentation.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_redmineProjectID.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	public void checkAttributeValue(PerunSessionImpl perunSession, Resource resource, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+		String id = (String) attribute.getValue();
+		if (id == null) {
+			throw new WrongAttributeValueException(attribute, resource, "Attribute can't be empty.");
+		}
+
+		Pattern pattern = Pattern.compile("^[a-z][-_a-z0-9]+$");
```
Comment (licehammer) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_redmineProjectID.java]: Can be moved outside the method for performance improvement.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_redmineProjectID.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		Matcher match = pattern.matcher(id);
+
+		if (!match.matches()) {
+			throw new WrongAttributeValueException(attribute, resource, "Bad format of attribute redmineProjectID.");
```
Comment (licehammer) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_redmineProjectID.java]: The exception message will be shown to end users. Please add some information what is the correct format to the message.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_redmineProjectID.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+		}
+	}
+
```
Comment (licehammer) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_redmineProjectID.java]: Remove trailing whitespace.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_redmineProjectID.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (licehammer) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_redmineProjectID.java]: Delete this comment.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_redmineProjectID.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+ * Module for redmine project ID
+ *
+ * @author Peter Balcirak <peter.balcirak@gmail.com>
+ * @date 29.3.2016
+ */
+public class urn_perun_resource_attribute_def_def_redmineProjectID extends ResourceAttributesModuleAbstract implements ResourceAttributesModuleImplApi {
+
+        private static final Pattern pattern = Pattern.compile("^[a-z][-_a-z0-9]+$");
```
Comment (licehammer) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_redmineProjectID.java]: Use tabs for indentation, not spaces.

**Comment by balcirakpeter** [comment]
Problem is solving in #1154

**Pull request closed without merging**